src/languages/typescript.ts

```diff
@@ -80,6 +86,25 @@ function valueMatcher() {
   );
 }
 
+export function typeSelectionExtractor(
+  editor: TextEditor,
+  node: SyntaxNode
+): SelectionWithContext {
+  const previousNode = node.previousSibling;
+
+  const leadingDelimiterRange =
+    previousNode != null && previousNode.type === ":"
+      ? makeRangeFromPositions(previousNode.startPosition, node.startPosition)
+      : null;
+
+  return {
+    ...simpleSelectionExtractor(editor, node),
+    context: {
+      leadingDelimiterRange,
+    },
+  };
+}
+
```

Comment:
How come the default one didn't work?

Reply:
The default implementation assumes that the name/key is a previous sibling to the type. That is true for python but not typescript according to the tree sitter.
I have some ideas for rewriting that entire thing to work for both when I have the time.

Reply:
Ah right ok. Fine with having it here for now